packages/@ourworldindata/grapher/src/stackedCharts/StackedBarChart.tsx

```diff
@@ -185,7 +185,9 @@ export class StackedBarChart
     }
 
     @computed protected get paddingForLegendRight(): number {
-        return this.showHorizontalLegend ? 0 : this.sidebarWidth + 20
+        return this.showHorizontalLegend
+            ? 0
+            : this.sidebarWidth + (this.sidebarWidth > 0 ? 20 : 0)
```

Comment:
I find this clearer:
```suggestion
            : (this.sidebarWidth > 0 ? this.sidebarWidth + 20 : 0)
```